### ISSUE TITLE
Immediate Search (Incremental search) Disable option

### ISSUE THREAD

**Issue by ajbelle** [JabRef/jabref#1034]
* JabRef version 3.0 onwards including 3.2
* Win7 Enterprise
* Searching large 45Mb .bib file:
   Result = JabRef immediately searches, blocking further typing into the search box
  Apparent  Reason = Large file takes time to search so this interrupts typing until it frees up.
   User Experience:  Very annoyingly, and time wasting . 
   Why in Stable Release: On a small test.bib file this delay is not noticeable, so it was not picked up during debugging.

On older JabRef versions that had Search in the LHS panel, I turned off `Incremental search` and the problem went away, but this disable functionality has been removed!  Discussion can be found at https://github.com/JabRef/jabref/issues/104 @tobiasdiez suggested I create this issue. According to @tobiasdiez the issue is separate, but turning off `Incremental search` used to fix it.

**Comment by stefan-kolb** [comment]
We should maybe do some small benchmarks how fast our search is and if it can/needs to be improved.

**Comment by oscargus** [comment]
These should be the critical lines of code though:

``` java
        // Search if user press enter
        searchField.addActionListener(e -> performSearch());

        // Subscribe to changes to the text in the search field in order to "live search"
        JTextFieldChangeListenerUtil.addChangeListener(searchField, e -> performSearch());
```

or rather, adding a parameter to performSearch which determines if a checkbox should be checked before searching.

**Comment by oscargus** [comment]
There's a tentative version available for download in about 20 minutes at http://builds.jabref.org/livesearch

(@ajbelle, this is based on the most recent version of JabRef, still I do not think it should break your massive file links...)

**Comment by ajbelle** [comment]
Thank you EASTER BUNNIES @oscargus and Co. I have been using 3.3dev-snapshot-2016-03-25-livesearch-55fc5f2 for several hours and it is working great. The "search on typing" icon is exactly what was required :-) Could I suggest the case sensitive icon would be more intuitive if one of the letters was capitalised, say `aBc` 
My repository is nearing 30 000 records and the diference is noticable on a Corei7 Win7 box.

**Comment by oscargus** [comment]
Good to hear that it worked! There is this icon which is meant to be used for case sensitive:

![case-sensitive-alt](https://cloud.githubusercontent.com/assets/8114497/14059572/7e0864cc-f345-11e5-9605-f9666c0b7500.png)

Not sure if it is much better. Makes very much sense if you are English speaking, maybe not so much otherwise. (We are trying to use provided icons.) The text indicates that this is an alternative version of another icon, but I cannot seem to find it.

**Comment by ajbelle** [comment]
That would be easily recognised by anyone using a Roman based letter set. It took me a while to realise that was a suit 'case' :-) Most Asians and Middle easterns I know using JabRef know more than enough English and the basic keyboard that I doubt anyone would complain. Maybe people still using Runes or Ogham script may struggle LOL. THX again.

**Comment by simonharrer** [comment]
Regarding the case sensitive icon, I created an issue so that a better icon will be created in the future in our icon font: https://github.com/Templarian/MaterialDesign/issues/1100

**Comment by simonharrer** [comment]
@ajbelle how many entries do you have in your 45mb file? I have created a JabRef version with a very fast search. Can you try it out if this would solve your problem? http://builds.jabref.org/fast-search/ In my tests, searching with 40k entries was acceptable on my machine, with a slight delay for warmup. Btw. filter is faster than float.

**Comment by ajbelle** [comment]
THAT IS GREAT. Mine is 30k entries (size is given in the question below), but many are lengthy and include abstracts ...   My problem with V3.2 was that it locked up typing entry long enough to mess it up and be annoying. The search time itself was not too bad. Your fix has solved that issue, and the search is lightning fast.

A side question, my file size dropped from 50MB to about 45MB when I moved to Ver3.2. In Notepad++  line count dropped from 512634 to 483231. I could not spot what had changed and assumed it was a better parsing ‘clearing out’ dead lines, however in this current testing I reverted back to Ver2.1 and it shot back up!!!! 
What are the extra lines, and how they are no longer needed?  Should I post this question to discussion? I couldn't find any info on this.
